wasm/src/main/scala/wasm4s/WasmContext.scala

```diff
@@ -208,6 +218,32 @@ trait TypeDefinableWasmContext extends ReadOnlyWasmContext { this: WasmContext =
     if (typeRef.dimensions > 1) IRTypes.ArrayType(typeRef.copy(dimensions = typeRef.dimensions - 1))
     else inferTypeFromTypeRef(typeRef.base)
   }
+
+  /** http://simonsapin.github.io/wtf-8/#encoding-ill-formed-utf-16
+    */
+  private def encodeStringToWTF16LE(input: String): Array[Byte] = {
+    val result = scala.collection.mutable.ArrayBuffer[Int]()
+    var i = 0
+    while (i < input.length) {
+      val codePoint = input.codePointAt(i)
+      if (codePoint < 0x10000) {
+        // BMP code point
+        result += codePoint
+        i += Character.charCount(codePoint)
+      } else {
+        // Supplementary code point
+        val highSurrogate = ((codePoint - 0x10000) >> 10) + 0xD800
+        val lowSurrogate = ((codePoint - 0x10000) & 0x3FF) + 0xDC00
+        result += highSurrogate
+        result += lowSurrogate
+        i += 2
+      }
+    }
```

Comment:
You made things difficult for yourself, but OK. I'm pretty sure this is equivalent to
```scala
val result = input.toCharArray()
```